Sources/web3swift/Utils/EIP/EIP712/EIP712Parser.swift

```diff
@@ -0,0 +1,297 @@
+
+        func encodeField(_ field: EIP712TypeProperty,
+                         value: AnyObject?) throws -> (encTypes: [ABI.Element.ParameterType], encValues: [Any]) {
+            var encTypes: [ABI.Element.ParameterType] = []
+            var encValues: [Any] = []
+            if field.type == "string" {
+                guard let value = value as? String else {
+                    throw Web3Error.processingError(desc: "EIP712Parser. Type metadata '\(field)' and actual value '\(String(describing: value))' type doesn't match. Cannot cast value to String.")
+                }
+                encTypes.append(.bytes(length: 32))
+                encValues.append(value.sha3(.keccak256).addHexPrefix())
+            } else if field.type == "bytes"{
+                guard let value = value as? Data else {
+                    throw Web3Error.processingError(desc: "EIP712Parser. Type metadata '\(field)' and actual value '\(String(describing: value))' type doesn't match. Cannot cast value to Data.")
+                }
+                encTypes.append(.bytes(length: 32))
+                encValues.append(value.sha3(.keccak256))
+            } else if field.isArray {
+                guard let values = value as? [AnyObject] else {
+                    throw Web3Error.processingError(desc: "EIP712Parser. Custom type metadata '\(field)' and actual value '\(String(describing: value))' type doesn't match. Cannot cast value to [AnyObject].")
+                }
+                encTypes.append(.bytes(length: 32))
+                let subField = EIP712TypeProperty(name: field.name, type: field.coreType)
+                var encodedSubTypes: [ABI.Element.ParameterType] = []
+                var encodedSubValues: [Any] = []
+                try values.forEach { value in
+                    let encoded = try encodeField(subField, value: value)
+                    encodedSubTypes.append(contentsOf: encoded.encTypes)
+                    encodedSubValues.append(contentsOf: encoded.encValues)
+                }
+
+                guard let encodedValue = ABIEncoder.encode(types: encodedSubTypes, values: encodedSubValues) else {
+                    throw Web3Error.processingError(desc: "EIP712Parser. Failed to encode an array of custom type. Field: '\(field)'; value: '\(String(describing: value))'.")
+                }
+
+                encValues.append(encodedValue.sha3(.keccak256))
+            } else if types[field.coreType] != nil  {
+                encTypes.append(.bytes(length: 32))
+                if let value = value as? [String : AnyObject] {
+                    encValues.append(try encodeData(field.type, data: value).sha3(.keccak256))
+                } else {
+                    encValues.append(Data(count: 32))
+                }
+            } else {
+                encTypes.append(try ABITypeParser.parseTypeString(field.type))
+                encValues.append(value as Any)
+            }
+            return (encTypes, encValues)
+        }
```

Comment:
Would you please consider to split this one up to fourth separated yet serially called functioning sake of code clarity?

Those are: `213:220`, `222:226`, `239:256`, `256:263` and the last one the whole `if else` chaining.